### PR TITLE
Split libats and bring libats-messaging

### PR DESCRIPTION
**Pull request by simao**

### CODE REVIEW

**Comment by simao** [comment]
@Danten @CalumMcCall 

Just a quick try at spliting libats and bringing libats-messaging here.

Still using an older version of cats/circe, but maybe we can build on this and upgrade it here?

**Comment by simao** [comment]
Just added a commit to upgrade circe + cats, this should be tested when integrating this library

**Comment by Danten** [comment]
Cool stuff, shouldn't the package name change names? They shouldn't be named `org.genivi.sota....`

**Comment by simao** [comment]
yes!

File: libats-messaging/src/main/scala/org/genivi/sota/messaging/kafka/MessageListener.scala [hunk 1]
```diff
@@ -0,0 +1,60 @@
+package org.genivi.sota.messaging.kafka
```
Comment (Danten) [inline libats-messaging/src/main/scala/org/genivi/sota/messaging/kafka/MessageListener.scala]: This file should probably not be in a `kafka` package

**Pull request merged**